openhexa/sdk/pipelines/parameter.py

```diff
@@ -125,7 +132,179 @@ def normalize(value: typing.Any) -> typing.Any:
+        return GCSConnectionType
+
+    @property
+    def accepts_choice(self) -> bool:
+        return False
+
+    @property
+    def accepts_multiple(self) -> bool:
+        return False
+
+    def validate(self, value: typing.Optional[typing.Any], *, allow_empty: bool = True) -> typing.Optional[str]:
+        if not allow_empty and value == "":
+            raise ParameterValueError("Empty values are not accepted.")
+
+        if not isinstance(value, str):
+            raise ParameterValueError(f"Invalid type for value {value} (expected {str}, got {type(value)})")
+
+        return value
+
+
+class DHIS2ConnectionType(ParameterType):
+    @property
+    def spec_type(self) -> str:
+        return "dhis2"
+
+    @property
+    def expected_type(self) -> typing.Type:
+        return DHIS2ConnectionType
+
+    @property
+    def accepts_choice(self) -> bool:
+        return False
+
+    @property
+    def accepts_multiple(self) -> bool:
+        return False
+
+    def validate(self, value: typing.Optional[typing.Any], *, allow_empty: bool = True) -> typing.Optional[str]:
+        if not allow_empty and value == "":
+            raise ParameterValueError("Empty values are not accepted.")
+
+        if not isinstance(value, str):
+            raise ParameterValueError(f"Invalid type for value {value} (expected {str}, got {type(value)})")
+
+        return value
+
+
+class IASOConnectionType(ParameterType):
+    @property
+    def spec_type(self) -> str:
+        return "iaso"
+
+    @property
+    def expected_type(self) -> typing.Type:
+        return IASOConnectionType
+
+    @property
+    def accepts_choice(self) -> bool:
+        return False
+
+    @property
+    def accepts_multiple(self) -> bool:
+        return False
+
+    def validate(self, value: typing.Optional[typing.Any], *, allow_empty: bool = True) -> typing.Optional[str]:
+        if not allow_empty and value == "":
+            raise ParameterValueError("Empty values are not accepted.")
+
+        if not isinstance(value, str):
+            raise ParameterValueError(f"Invalid type for value {value} (expected {str}, got {type(value)})")
+
+        return value
+
+
+class CustomConnectionType(ParameterType):
+    @property
+    def spec_type(self) -> str:
+        return "custom"
+
+    @property
+    def expected_type(self) -> typing.Type:
+        return str
+
+    @property
+    def accepts_choice(self) -> bool:
+        return False
+
+    @property
+    def accepts_multiple(self) -> bool:
+        return False
+
+    def validate(self, value: typing.Optional[typing.Any], *, allow_empty: bool = True) -> typing.Optional[str]:
+        if not allow_empty and value == "":
+            raise ParameterValueError("Empty values are not accepted.")
+
+        if not isinstance(value, str):
+            raise ParameterValueError(f"Invalid type for value {value} (expected {str}, got {type(value)})")
+
+        return super().validate(value, allow_empty)
+
+
+TYPES_BY_PYTHON_TYPE = {
```

Comment:
Do you think that it would be a lot of added work to rename `String` to `StringType`, `Boolean` to `BooleanType` and so on?

Reply:
Let me try